modules/display.js

```diff
@@ -53,10 +56,27 @@ const display = () => {
   checkboxes.forEach((checkbox) => {
     checkbox.addEventListener('change', () => {
       const li = checkbox.closest('section').querySelector('li');
+      let i = 0;
       if (checkbox.checked) {
         li.classList.add('completed');
+        const liId = parseInt(li.getAttribute('data-index'), 10);
+        while (i < itemData.length) {
+          if (liId === itemData[i].index) {
+            itemData[i].completed = true;
+            localStorage.setItem('todoListItems', JSON.stringify(itemData));
+          }
+          i += 1;
+        }
       } else {
         li.classList.remove('completed');
+        const liId = parseInt(li.getAttribute('data-index'), 10);
+        while (i < itemData.length) {
+          if (liId === itemData[i].index) {
+            itemData[i].completed = false;
+            localStorage.setItem('todoListItems', JSON.stringify(itemData));
+          }
+          i += 1;
+        }
       }
```

Comment:
Well done in this milstone , but to avoid redendency you can mark the checkboxes as checked in the same loop where you create elements in the dom "the first loop" if complete attribute on an element is true mark it as checked